units/host_monitoring_last_container_finish_time.go

```diff
@@ -0,0 +1,77 @@
+package units
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+)
+
+const (
+	lastContainerFinishTimeJobName = "last-container-finish-time"
+)
+
+func init() {
+	registry.AddJobType(lastContainerFinishTimeJobName, func() amboy.Job {
+		return makeIdleHostJob()
+	})
+
+}
+
+type lastContainerFinishTimeJob struct {
+	job.Base   `bson:"metadata" json:"metadata" yaml:"metadata"`
+	Terminated bool `bson:"terminated" json:"terminated" yaml:"terminated"`
+
+	env      evergreen.Environment
+	settings *evergreen.Settings
+}
+
+func makeLastContainerFinishTimeJob() *lastContainerFinishTimeJob {
+	j := &lastContainerFinishTimeJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    lastContainerFinishTimeJobName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+
+	return j
+}
+
+func NewLastContainerFinishTimeJob(env evergreen.Environment, id string) amboy.Job {
```

Comment:
The env param can be removed from here and also PopulateLastContainerFinishTimeJobs